src/ui/public/render_directive/render_directive.js

```diff
@@ -6,7 +6,8 @@ module.directive('renderDirective', function () {
   return {
     restrict: 'E',
     scope: {
-      'definition': '='
+      'definition': '=',
+      'scope': '=?'
```

Comment:
I took a stab at this in https://github.com/elastic/kibana/pull/6115, mind trying it out?